li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -215,7 +224,16 @@ private LiKafkaFederatedConsumerImpl(LiKafkaConsumerConfig configs, MetadataServ
       // We assume that such information will be kept and used by the metadata service client itself.
       //
       // TODO: make sure this is not blocking indefinitely and also works when Mario is not available.
-      _mdsClient.registerFederatedClient(this, _clusterGroup, configs.originals(), _mdsRequestTimeoutMs);
+      boolean registerSuccessful = _mdsClient.registerFederatedClient(this, _clusterGroup, configs.originals(), _mdsRequestTimeoutMs);
```

Comment:
i dislike the fact that success/failure is only communicated via a boolean here - can we distinguish between "cant find/access conductor (which would be a warning) and ssl handshake error here? maybe we should consider exceptions instead of a boolean here.

Reply:
I'm not sure how to use exceptions here since we are submitting the register job to an executor and check if the state has been changed to CONNECTED within the specified timeout. If we want to differentiate the different error cases, we need to instrument conductor client to throw different exceptions upon connecting to conductor server and all the way up to federated client.